lib/cocoapods/validator.rb

```diff
@@ -662,7 +666,19 @@ def build_pod
           if scheme.nil?
             UI.warn "Skipping compilation with `xcodebuild` because target contains no sources.\n".yellow
           else
-            output = xcodebuild('build', scheme, 'Release')
+            if analyze
+              output = xcodebuild('analyze', scheme, 'Release')
+              find_output = Executable.execute_command('find', [validation_dir, '-name', '*.html'], false)
```

Comment:
Is this a reliable way to determine whether analysis failed? Is there a reason why the `xcodebuild` call won't itself fail?

Reply:
Unfortunately not, the `xcodebuild` succeeds if the build succeeds even if there are `analyze` issues. See the discussion at https://stackoverflow.com/a/39635929.

Reply:
should we move this into an `analyze_pod` method for consistency with `build_pod` and `test_pod`?

Reply:
I'm not sure, since analyze is mostly an option to build_pod.

Reply:
Gotcha makes sense